api/src/scripts/populate_db.py

```diff
@@ -232,6 +226,50 @@ def add_entity(entity, priority):
             )
             add_entity(mdb_external_id, 4)
 
+        # Iterate again over the contents of the csv files to process the feed references.
+        for index, row in self.df.iterrows():
+            mdb_id = f"mdb-{int(row['mdb_source_id'])}"
+            feed = feed_map[mdb_id]
+            if row["data_type"] == "gtfs_rt":
+                # Feed Reference
+                if row["static_reference"] is not None:
+                    static_reference_mdb_id = f"mdb-{int(row['static_reference'])}"
+                    referenced_feed = feed_map.get(static_reference_mdb_id, None)
+                    if referenced_feed:
+                        referenced_feed.gtfs_rt_feeds.append(feed)
+
+            # Process redirects
+            raw_redirects = row.get("redirect.id", None)
+            redirects_ids = raw_redirects.split("|") if raw_redirects is not None else []
+            raw_comments = row.get("redirect.comment", None)
+            comments = raw_comments.split("|") if raw_comments is not None else []
+
+            if len(redirects_ids) != len(comments):
+                self.logger.warn(f'Number of redirect ids and redirect comments differ for feed {mdb_id}')
+
+            for mdb_source_id in redirects_ids:
+                if len(mdb_source_id) == 0:
+                    # since there is a 1:1 correspondence between redirect ids and comments, skip also the comment
+                    comments = comments[1:]
+                    continue
+                if comments:
+                    comment = comments.pop(0)
+                else:
+                    comment = ""
+
+                target_stable_id = f"mdb-{mdb_source_id}"
+                target_feed = feed_map.get(target_stable_id, None)
+
+                if target_feed:
```

Comment:
For safety, we should prevent a recursive redirection (feed.id == target_feed.id)

Reply:
Added a first level check (feed = redirect feed) but we should also check for indirect recursion (feed1 -> feed2 -> feed1)